test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -239,8 +249,23 @@ struct test_transform_iterator {
         test_transform_effect(in1.begin(),  in1.end(),  in2.begin());
         test_transform_effect(in1.cbegin(), in1.cend(), in2.begin());
 
-        auto new_transform_iterator = oneapi::dpl::make_transform_iterator(in2.begin(), [](T2& x) { return x + 1; });
-        test_random_iterator(new_transform_iterator);
+        transform_functor new_functor;
+        oneapi::dpl::transform_iterator<typename ::std::vector<T1>::iterator, transform_functor> _it1(in1.begin());
+        oneapi::dpl::transform_iterator<typename ::std::vector<T1>::iterator, transform_functor> _it2(in1.begin(), new_functor);
+
+        ::std::forward_list<int> f_list{1,2,3,4,5,6};
+        oneapi::dpl::transform_iterator<decltype(f_list.begin()),transform_functor> list_it1(f_list.begin(),new_functor);
+        oneapi::dpl::transform_iterator<decltype(f_list.begin()), transform_functor> list_it2(f_list.end(),new_functor);
+        int sum=0;
+        ::std::for_each(list_it1,list_it2,[&sum](auto  x){sum+=x;});
+        EXPECT_TRUE(sum==27,"wrong result from for_each with forward_iterator wrapped with transform_iterator");
+
+        auto test_lambda = [](T2& x){ return x+1; };
+        auto new_transform_iterator = oneapi::dpl::make_transform_iterator(in2.begin(), test_lambda);
+        EXPECT_TRUE(_it1.base() == in1.begin(), "wrong result from transform_iterator::base");
+        static_assert(::std::is_same<decltype(new_transform_iterator.functor()),decltype(test_lambda)>::value,
+            "wrong result from transform_iterator::functor");
+        test_random_iterator(_it2);
```

Comment:
Overall (not only for this part of the code). Please keep in mind spaces. The code looks hardly readable to me. Please put spaces after commas and before/after operators like `=`, `+`, etc.

Reply:
Aligned